modules/cudaoptflow/src/nvidiaOpticalFlow.cpp

```diff
@@ -656,18 +679,602 @@ void NvidiaOpticalFlowImpl::upSampler(InputArray _flow, int width, int height,
+
+    if (m_inputStream || m_outputStream)
+    {
+        NVOF_API_CALL(GetAPI()->nvOFSetIOCudaStreams(GetHandle(),
+            StreamAccessor::getStream(m_inputStream), StreamAccessor::getStream(m_outputStream)));
+    }
+
+    //Input Buffer 1
+    NVOF_API_CALL(GetAPI()->nvOFCreateGPUBufferCuda(GetHandle(),
+        &m_inputBufferDesc, NV_OF_CUDA_BUFFER_TYPE_CUDEVICEPTR, &m_hInputBuffer));
+    m_frame0cuDevPtr = GetAPI()->nvOFGPUBufferGetCUdeviceptr(m_hInputBuffer);
+    NVOF_API_CALL(GetAPI()->nvOFGPUBufferGetStrideInfo(
+        m_hInputBuffer, &m_inputBufferStrideInfo));
+
+    //Input Buffer 2
+    NVOF_API_CALL(GetAPI()->nvOFCreateGPUBufferCuda(GetHandle(),
+        &m_inputBufferDesc, NV_OF_CUDA_BUFFER_TYPE_CUDEVICEPTR, &m_hReferenceBuffer));
+    m_frame1cuDevPtr = GetAPI()->nvOFGPUBufferGetCUdeviceptr(m_hReferenceBuffer);
+    NVOF_API_CALL(GetAPI()->nvOFGPUBufferGetStrideInfo(
+        m_hReferenceBuffer, &m_referenceBufferStrideInfo));
+
+    //Output Buffer
+    NVOF_API_CALL(GetAPI()->nvOFCreateGPUBufferCuda(GetHandle(),
+        &m_outputBufferDesc, NV_OF_CUDA_BUFFER_TYPE_CUDEVICEPTR, &m_hOutputBuffer));
+    m_flowXYcuDevPtr = GetAPI()->nvOFGPUBufferGetCUdeviceptr(m_hOutputBuffer);
+    NVOF_API_CALL(GetAPI()->nvOFGPUBufferGetStrideInfo(
+        m_hOutputBuffer, &m_outputBufferStrideInfo));
+
+    if (m_scaleFactor > 1)
+    {
+        m_outputBufferDesc.width = (m_width + m_gridSize - 1) / m_gridSize;;
+        m_outputBufferDesc.height = (m_height + m_gridSize - 1) / m_gridSize;;
+
+        //Output UpScaled Buffer
+        NVOF_API_CALL(GetAPI()->nvOFCreateGPUBufferCuda(GetHandle(),
+            &m_outputBufferDesc, NV_OF_CUDA_BUFFER_TYPE_CUDEVICEPTR, &m_hOutputUpScaledBuffer));
+        m_flowXYUpScaledcuDevPtr = GetAPI()->nvOFGPUBufferGetCUdeviceptr(m_hOutputUpScaledBuffer);
+        NVOF_API_CALL(GetAPI()->nvOFGPUBufferGetStrideInfo(
+            m_hOutputUpScaledBuffer, &m_outputUpScaledBufferStrideInfo));
+    }
+
+    //Hint Buffer
+    if (m_enableExternalHints)
+    {
+        NVOF_API_CALL(GetAPI()->nvOFCreateGPUBufferCuda(GetHandle(),
+            &m_hintBufferDesc, NV_OF_CUDA_BUFFER_TYPE_CUDEVICEPTR, &m_hHintBuffer));
+        m_hintcuDevPtr = GetAPI()->nvOFGPUBufferGetCUdeviceptr(m_hHintBuffer);
+        NVOF_API_CALL(GetAPI()->nvOFGPUBufferGetStrideInfo(
+            m_hHintBuffer, &m_hintBufferStrideInfo));
+    }
+
+    //Cost Buffer
+    if (m_enableCostBuffer)
+    {
+        NVOF_API_CALL(GetAPI()->nvOFCreateGPUBufferCuda(GetHandle(),
+            &m_costBufferDesc, NV_OF_CUDA_BUFFER_TYPE_CUDEVICEPTR, &m_hCostBuffer));
+        m_costcuDevPtr = GetAPI()->nvOFGPUBufferGetCUdeviceptr(m_hCostBuffer);
+        NVOF_API_CALL(GetAPI()->nvOFGPUBufferGetStrideInfo(
+            m_hCostBuffer, &m_costBufferStrideInfo));
+    }
+}
+
+void NvidiaOpticalFlowImpl_2::calc(InputArray _frame0, InputArray _frame1, InputOutputArray _flow,
+    Stream& stream, InputArray hint, OutputArray cost)
+{
```

Comment:
> `cudaoptflow/src/nvidiaOpticalFlow.cpp:993:13: warning: unused parameter 'stream' [-Wunused-parameter]`

Please use `CV_UNUSED(stream);` if this parameter is not needed.

Reply:
Done

Reply:
Done